### DISCUSSION TITLE
How to use with other services

### DISCUSSION THREAD

**Discussion by thomas169** [afpineda/NuS-NimBLE-Serial#3] [General]
Hi, thanks for the library first off. In the readme it is mentioned you can use NuSerial alongside any other services you have, but its not demo'd in any of the examples and from what I see the pServer used by the library is private so not accessible. Is it just a stale readme or is it actually possible to use lib with other services as mentioned. If the latter do you have any pointers on how to do this?

**Comment by afpineda** [comment]
Thanks for your support.

In order to create and use other GATT services, just use the [NimBLE API](https://h2zero.github.io/NimBLE-Arduino/index.html). Ignore the fact that NuS-Nimble-Serial is there. In particular, you get the GATT server with [NimBLEDevice::getServer() or  NimBLEDevice::createServer()](https://h2zero.github.io/NimBLE-Arduino/class_nim_b_l_e_device.html#a41d4cc7c85cba0721c6649b001b31b4d). Then, you build your services from there. 
The code of `NordicUARTService::init()` works as an example of that.
